test/application.py

```diff
@@ -92,6 +92,19 @@ def GET(self, name):
         response = app.request('/b/foo?x=2')
         self.assertEquals(response.status, '301 Moved Permanently')
         self.assertEquals(response.headers['Location'], 'http://0.0.0.0:8080/hello/foo?x=2')
+
+    def test_routing(self):
+        urls = (
+            "/foo", "foo"
+        )
+
+        class foo:
+            def GET(self):
+                return "foo"
+
+        app = web.application(urls, {"foo": foo})
+
+        self.assertEquals(app.request('/foo\n').data, b'not found')
```

Comment:
Is there a test that covers the good case? e.g. self.assertEquals(app.request('/foo').data, 'foo') ?